xgi/generators/simplicial_complexes.py

```diff
@@ -130,26 +130,31 @@ def flag_complex(G, max_order=2, ps=None, seed=None):
         random.seed(seed)
 
     nodes = G.nodes()
+    N = len(nodes)
     edges = G.edges()
 
-    # compute all maximal cliques to fill
-    max_cliques = list(nx.find_cliques(G))
+    # compute all cliques to fill
+    if max_order is None:
+        cliques_to_add = list(nx.find_cliques(G)) # max cliques
+    else: # avoid adding many unnecessary redundant cliques
+        cliques_to_add = []
+        for clique in nx.enumerate_all_cliques(G): # sorted by size
+            if len(clique)<=max_order+1: 
+                cliques_to_add.append(clique)
+            else:
+                break # dont go over whole list if not necessary
+        cliques_to_add = cliques_to_add[N:] # remove singletons
```

Comment:
```suggestion
            if 1 < len(clique) <= max_order+1: 
                cliques_to_add.append(clique)
            else:
                break # dont go over whole list if not necessary
```

Reply:
I thought of that too but it does not work because of the `break`.. first cliques are singletons and the results list if empty

Reply:
Replaced with an if at the top